compiler+runtime/include/cpp/jank/read/parse.hpp

```diff
@@ -12,33 +14,75 @@ namespace jank::runtime
 /* TODO: Rename file to processor. */
 namespace jank::read::parse
 {
-  static option<char> get_char_from_literal(native_persistent_string const &sv)
+  static option<native_persistent_string>
+  parse_character_in_base(native_persistent_string_view const &char_literal, int const base)
+  {
+    try
+    {
+      size_t chars_processed{};
+      auto const codepoint(
+        std::stol(native_persistent_string{ char_literal.data() + 2, char_literal.size() - 2 },
+                  &chars_processed,
+                  base));
+
+      /* Some characters that weren't processed at all.
+       * An example would be when trying to parse `12a` in octal base,
+       * `stol` will only parse `12` and will ignore `a`, returning `2`
+       * in `chars_processed`.
+       * We return none in that case, to represent to the downstream callers
+       * the enability to parse the char literal.
+       *
+       * Refer: https://en.cppreference.com/w/cpp/string/basic_string/stol
+       */
+      if(chars_processed != char_literal.size() - 2)
+      {
+        return none;
+      }
+
+      std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> converter;
```

Comment:
The `converter` has a member fn on it to check how many codepoints were consumed. We could check that this is `1` and fail otherwise. Also, we may be getting into an area here where `result` is more useful than `option`.

Reply:
Is it because we also want to return the cause of the error?

If so, what's the purpose of returning cause of the error? Is it to show a better error message?

Reply:
Yes and yes!